### PR TITLE
Implented database authentication for users registration

### PR DESCRIPTION
**Pull request by abhishek-vasyerp**
changes in this commit:
- added database schema to store credentials
- implemented user registration functionality to store user's information

### CODE REVIEW

File: src/main/java/com/vasy/security/controller/RoleController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	
+	
+	@GetMapping("/register")
+	public ModelAndView register() {
+		ModelAndView modelAndView = new ModelAndView("RegisterPage");
+		return modelAndView;
+	}
+	
+	
+	@PostMapping("/addUser")
```
Comment (rahulm0901) [inline src/main/java/com/vasy/security/controller/RoleController.java]: end point is not proper, it must be add-user

File: src/main/java/com/vasy/security/dto/UserDetailDto.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.vasy.security.dto;
+
+import lombok.Data;
+
+@Data
+public class UserDetailDto {
+	
+	private long id;
```
Comment (rahulm0901) [inline src/main/java/com/vasy/security/dto/UserDetailDto.java:8]: I Id is auto generated then why are you taking as input?

File: src/main/java/com/vasy/security/service/UserAndAuthoritiesServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		// TODO Auto-generated method stub
+
+		Users user = new Users();
+		user.setUsername(userDetailDto.getUsername());
+		String encryptedPass = passwordEncoder.encode(userDetailDto.getPassword());
+		user.setPassword(encryptedPass);
+		user.setEnabled(true);
+		 long roleId = Long.parseLong( userDetailDto.getAuthority());
+		
+		Authorities authorities=this.authoritiesRepository.findById(roleId).orElseThrow(() -> new Exception("No found"));
```
Comment (rahulm0901) [inline src/main/java/com/vasy/security/service/UserAndAuthoritiesServiceImpl.java]: There must be specific exception to be thrown, like RoleNotFound Exception.

File: src/main/java/com/vasy/security/dto/UserDetailDto.java (hunk 2 above)
Comment (abhishek-vasyerp) [reply src/main/java/com/vasy/security/dto/UserDetailDto.java:8]: I'm not taking id as input just passing in the response.

File: src/main/java/com/vasy/security/security/MyUniqueAuthenticationHandler.java [hunk 4]
```diff
@@ -16,17 +16,21 @@ public class MyUniqueAuthenticationHandler implements AuthenticationSuccessHandl
...
+                                        Authentication authentication) throws IOException, ServletException {
+
+        String role = authentication.getAuthorities().iterator().next().getAuthority();
+
+        System.out.println(role);
+
         List<GrantedAuthority> authorities = new ArrayList<>(authentication.getAuthorities());
         for (GrantedAuthority authority : authorities) {
-            if (authority.getAuthority().equals("ROLE_ADMIN")) {
+            if (authority.getAuthority().equals("ADMIN")) {
```
Comment (rahulm0901) [inline src/main/java/com/vasy/security/security/MyUniqueAuthenticationHandler.java:27]: Move this String Literals to constants

File: src/main/java/com/vasy/security/service/UserAndAuthoritiesServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		// TODO Auto-generated method stub
+
+		Users user = new Users();
+		user.setUsername(userDetailDto.getUsername());
+		String encryptedPass = passwordEncoder.encode(userDetailDto.getPassword());
+		user.setPassword(encryptedPass);
+		user.setEnabled(true);
+		 long roleId = Long.parseLong( userDetailDto.getAuthority());
+		
+		Authorities authorities=this.authoritiesRepository.findById(roleId).orElseThrow(() -> new RoleNotFoundException("No found"));
```
Comment (rahulm0901) [inline src/main/java/com/vasy/security/service/UserAndAuthoritiesServiceImpl.java:43]: Move this String Literals to constants

**Review by rahulm0901 — APPROVED** [review]

**Pull request merged**